src/plugins/intel_gpu/src/plugin/program_builder.cpp

```diff
@@ -318,6 +293,106 @@ ProgramBuilder::ProgramBuilder(cldnn::engine& engine, const ExecutionConfig& con
         m_networkOutputs = *outputs;
 }
 
+ProgramBuilder::ProgramBuilder(const std::shared_ptr<ov::Model>& model, cldnn::engine& engine,
+                                const ExecutionConfig& config,
+                                std::shared_ptr<ov::threading::IStreamsExecutor> task_executor)
+    : m_curBatch(-1)
+    , m_config(config)
+    , m_engine(engine)
+    , queryMode(false)
+    , m_task_executor(task_executor) {
+    if (m_task_executor == nullptr)
+        m_task_executor = cldnn::program::make_task_executor(m_config);
+
+    if (!model) {
+        OPENVINO_THROW("ov::Model pointer is nullptr");
+    }
+
+    // inner network is only allowed single batch
+    m_max_batch = 1;
+
+    // Set input / output for Parameter and Result in cldnn
+    InferenceEngine::InputsDataMap networkInputs = {};
+    InferenceEngine::OutputsDataMap networkOutputs = {};
+
+    auto create_data_for_result = [&](const ngraph::Output<ngraph::Node>& output,
```

Comment:
[random spot] Does https://github.com/openvinotoolkit/openvino/pull/18920 solves the original issue? If yes, then maybe we can postpone this PR a little bit, as you use/modify some things that are going to be removed

Reply:
@vladimir-paramuzov the main purpose of this PR is creating cldnn::program without CNNNetwork to fix zero dimension issue and I think https://github.com/openvinotoolkit/openvino/pull/18920 can solve the issue. Thus, I will postpone this PR until  https://github.com/openvinotoolkit/openvino/pull/18920 will be merged. thanks.

Reply:
@ahnyoung-paul  PR https://github.com/openvinotoolkit/openvino/pull/18920 was merged. So please update this one or close if it's not needed anymore.